drivers/sensor/sx9500/sx9500.c

```diff
@@ -120,23 +120,27 @@ int sx9500_init(const struct device *dev)
 		return -EINVAL;
 	}
 
-	if (sx9500_setup_interrupt(dev) < 0) {
-		LOG_DBG("sx9500: failed to setup interrupt");
-		return -EINVAL;
+	if (cfg->int_gpio.port) {
+		if (sx9500_setup_interrupt(dev) < 0) {
+			LOG_DBG("sx9500: failed to setup interrupt");
+			return -EINVAL;
```

Comment:
I know that is outside of the scope of that MR but I would rather propagate the return code from the `sx9500_setup_interrupt` instead of returning EINVAL which IMO. is not suitable here.

Reply:
Agree, lots of sensor drivers misuse error values in this way and it should be fixed in future PRs.

Reply:
agreed, this could be part of another round of cleanups. We should probably write this into coding guidelines as well.